src/hal/lib/hal_comp.c

```diff
@@ -372,36 +372,39 @@ const char *hal_comp_name(int comp_id)
 
 // use only for owner_ids of pins, params or functs
 // may return NULL if buggy using code
+// This was a bottleneck with rcomps validating pins
+// Move checks so that instances get checked first
+
 hal_comp_t *halpr_find_owning_comp(const int owner_id)
 {
-    hal_comp_t *comp = halpr_find_comp_by_id(owner_id);
-    if (comp != NULL)
-	return comp;  // legacy case: the owner_id refers to a comp
+hal_comp_t *comp;
 
-    // nope, so it better be an instance
+    // more likely to be instcomp, so test that first
     hal_inst_t *inst = halpr_find_inst_by_id(owner_id);
-    if (inst == NULL) {
-	HALERR("BUG: owner_id %d refers neither to a hal_comp_t nor an hal_inst_t",
-	       owner_id);
-	return NULL;
-    }
 
+    if (inst == NULL) { // is it a legacy comp?
+        comp = halpr_find_comp_by_id(owner_id);
+        if (comp != NULL)
```

Comment:
Not using curly braces after a conditional which only has a one line operation following it, has been accepted since the dawn of C.
Using curly braces takes an extra line (or two, depending upon your style) and litters the code with unnecessary braces to be traced back to figure out scopes.

Not using them when more than one line operation follows is just wrong, as only the first line will be executed as part of the conditional statement.  On that we are in full agreement.

Let's insert a full stop there.